PassGen/MainWindow.xaml.cs

```diff
@@ -1,74 +1,74 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Security.Cryptography;
 using System.Windows;
 
 namespace PassGen
 {
-    public partial class MainWindow : Window
+    public partial class MainWindow
     {
-        private RandomNumberGenerator rnd = RandomNumberGenerator.Create();
-        public MainWindow()
+        private static readonly RandomNumberGenerator randomNumberGenerator = RandomNumberGenerator.Create();
+
+        private static int GetNextNumber(int maxValue)
         {
-            InitializeComponent();
-            PLength.Minimum = 1;
-            PLength.Maximum = 40;
-            PLength.Value = 10;
+            var symbol = new byte[1];
+            randomNumberGenerator.GetBytes(symbol);
+            return symbol[0] % maxValue;
         }
 
-        private int getNewNumber(int mx)
+        public MainWindow()
         {
-            byte[] symbol = new byte[1];
-            rnd.GetBytes(symbol);
-            return symbol[0] % mx;
+            InitializeComponent();
         }
         
-        private void Button_Click(object sender, RoutedEventArgs e)
+        private void GenerateButtonClick(object sender, RoutedEventArgs e)
         {
-            string s = String.Empty;
-            for (int i = 0; i < PLength.Value; i++)
+            // for more info check ASCII table
+
+            var resultPassword = string.Empty;
+
+            const int startShift = ' ' + 1;
+            const int charRange = 'z' - '!' + 1;
+
+            for (var i = 0; i < PasswordLengthSlider.Value; i++)
             {
-                int xchar = 33 + getNewNumber(94);
-                if (Digits.IsChecked == true)
+                var charValue = startShift + GetNextNumber(charRange);
+
+                if (UseDigitsCheckBox.IsChecked == true)
                 {
-                    if (Symbols.IsChecked == false)
+                    if (UseSpecialSymbolsCheckBox.IsChecked == false)
                     {
-                        while (xchar < 48 || xchar > 57 && xchar < 65 || xchar > 90 && xchar < 97 || xchar > 122)
-                        {
-                            xchar = 33 + getNewNumber(94);
-                        }
+                        while (charValue < '0' || charValue > '9' && charValue < 'a' || charValue > 'z' && charValue < 'A' || charValue > 'Z')

```

Comment:
Broken loop, as `'a'` is bigger than `'A'`
Should be `charValue < '0' || charValue > '9' && charValue < 'A' || charValue > 'Z' && charValue < 'a' || charValue > 'z'`